staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -318,6 +312,10 @@ func (v *specStandardValidatorV3) validate(schema *apiextensions.JSONSchemaProps
 		allErrs = append(allErrs, field.Forbidden(fldPath.Child("dependencies"), "dependencies is not supported"))
 	}
 
+	if schema.Ref != nil {
```

Comment:
Does this mean we forbid schemata like described in https://spacetelescope.github.io/understanding-json-schema/structuring.html with local definitions? Then I do not agree with this PR.

Reply:
To my understanding the `id` of a document is only used as a base for relative references. With `#/definitions/foo` you can reference absolute definitions. The later is completely fine and users might use them. If we forbid that suddenly, we break compatibility.

Reply:
@sttts We don't support definitions now:

https://github.com/kubernetes/kubernetes/blob/23226c24d4fd11bcb7aafda128aecbe3de3a0732/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go#L313-L315

because OpenAPI v3 does not support definitions (https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#schemaObject)

`Ref` is used in OpenAPI v3 like this: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#reference-object.

Reply:
Alright. Poor OpenAPI (again). Then, without definitions my concerns are gone.

/unhold

Reply:
thanks for your review